src/core/a-entity.js

```diff
@@ -463,31 +509,52 @@ var proto = Object.create(ANode.prototype, {
     value: function (attr, value, componentPropValue) {
       var self = this;
       var component = this.components[attr] || components[attr];
-      var partialComponentData;
-      value = value === undefined ? '' : value;
       var componentObj = value;  // Deserialized value to send to the component.
       var componentStr = value;  // Serialized value to send to the DOM.
-      var oldValue;
+      // if there's a previous value it will be cached in the component.
+      var oldValue = this.components[attr] && this.components[attr].attr;
 
       if (component) {
         if (typeof value === 'string' && componentPropValue !== undefined) {
           // Update currently-defined component data with the new property value.
-          // Use native setAttribute in order not to double-parse properties.
-          partialComponentData = styleParser.parse(
-            HTMLElement.prototype.getAttribute.call(this, attr)) || {};
-          partialComponentData[value] = componentPropValue;
-          componentObj = partialComponentData;
+          // Use cached attributes to prevent unnecessary string parsings.
+          oldValue = oldValue ? utils.extend({}, oldValue) : {};
+          componentObj = component.attr || {};
+          componentObj[value] = componentPropValue;
         }
-        componentStr = component.stringify(componentObj);
+        self.setEntityAttribute(attr, oldValue, componentObj);
+        return;
       }
-      oldValue = this.getAttribute(attr);
       ANode.prototype.setAttribute.call(self, attr, componentStr);
-      self.setEntityAttribute(attr, oldValue, componentObj);
+      if (attr === 'mixin') { self.setEntityAttribute('mixin', undefined, componentObj); }
     },
     writable: window.debug
   },
 
   /**
+   * To make the DOM attributes reflect the state of the components.
+   *
+   * @param {bool} iterative - It calls updateDOM on the children
+   **/
+  updateDOM: {
```

Comment:
Maybe:

```js
flushToDom: {
  value: function (recurse)
```

I like `flush` since it builds the imagery of all of this data being built up that has been blocked from flowing to the DOM.